### PR TITLE
GEODE-7345: Decouple TcpServer from Core's Socket-Related Classes

### PR DESCRIPTION
**Pull request by Bill**
[GEODE-7345](https://issues.apache.org/jira/browse/GEODE-7345)

Eliminate these exceptions in `TcpServerDependenciesTest`:

```java
              // TODO - TCP socket related classes
              .or(type(SocketCreator.class))
              .or(type(SSLConfigurationFactory.class))
              .or(type(SecurableCommunicationChannel.class))
              .or(type(SocketCreatorFactory.class))
              .or(type(SSLConfigurationFactory.class))
```

This work is being done in pursuit of [GEODE-7343](https://issues.apache.org/jira/browse/GEODE-7343) which will break all dependencies from from the `org.apache.geode.distributed.internal.tcpserver` package to classes in `geode-core`. The end goal of that ticket is to move the `tcpserver` package into its own module, `geode-tcp-server` that will have no dependencies on `geode-core`. That new module will serve both `geode-core` and the eventual `geode-membership` module.

To eliminate the dependencies on `SocketCreator`, I defined a new interface, `TcpSocketCreator` and a factory method that turns a `SocketCreator` into one of those: `TcpSocketCreator.asTcpSocketCreator(SocketCreator)`.

The new `TcpSocketCreator` interface is a vastly narrowed (5 methods vs 50 methods) version of the `SocketCreator` class used throughout `geode-core`. This new interface has only the methods needed to server the needs of classes in the `tcserver` package (eventually the `geode-tcp-server` module).

In places where we were creating `SocketCreator` and using that to construct a `TcpClient` or `TcpServer`, we're now using `asTcpSocketCreator()` to adapt to the new (narrow) interface.

The other dependencies e.g. `SSLConfigiurationFactory` were easier to eliminate. I simply extracted construction of those objects up out of `TcpServer` and `TcpClient` as constructor parameters. Dependency injection wins.

### For all changes:
- [x] Is there a JIRA ticket associated with this PR? Is it referenced in the commit message?

- [x] Has your PR been rebased against the latest commit within the target branch (typically `develop`)?

- [ ] Is your initial contribution a single, squashed commit?

- [x] Does `gradlew build` run cleanly?

- [x] Have you written or updated unit tests to verify your changes?

- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?

### Note:
Please ensure that once the PR is submitted, check Concourse for build issues and
submit an update to your PR as soon as possible. If you need help, please send an
email to dev@geode.apache.org.

### CODE REVIEW

File: geode-core/src/main/java/org/apache/geode/distributed/internal/tcpserver/MembershipSocketCreator.java [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+ */
+
+package org.apache.geode.distributed.internal.tcpserver;
+
+import java.io.IOException;
+import java.net.InetAddress;
+import java.net.ServerSocket;
+import java.net.Socket;
+
+public interface MembershipSocketCreator {
```
Comment (upthewaterspout) [inline geode-core/src/main/java/org/apache/geode/distributed/internal/tcpserver/MembershipSocketCreator.java]: javadocs?

File: geode-core/src/main/java/org/apache/geode/internal/net/SocketCreatorFactory.java [hunk 2]
```diff
@@ -74,6 +80,48 @@ public static SocketCreator getSocketCreatorForComponent(
         sslConfigForComponent);
   }
 
+  public static class MembershipSocketCreatorAdapter implements MembershipSocketCreator {
```
Comment (upthewaterspout) [inline geode-core/src/main/java/org/apache/geode/internal/net/SocketCreatorFactory.java]: Make this top level?

File: geode-core/src/main/java/org/apache/geode/distributed/internal/tcpserver/MembershipSocketCreator.java (hunk 1 above)
Comment (Bill) [reply geode-core/src/main/java/org/apache/geode/distributed/internal/tcpserver/MembershipSocketCreator.java]: renamed to `TcpSocketCreator` and added a javadoc comment

File: geode-core/src/main/java/org/apache/geode/internal/net/SocketCreatorFactory.java (hunk 2 above)
Comment (Bill) [reply geode-core/src/main/java/org/apache/geode/internal/net/SocketCreatorFactory.java]: extracted as `SocketCreatorAdapter`

**Comment by Bill** [comment]
Fixed failing distributed tests.

**Comment by Bill** [comment]
Closing this PR while I fix this failing stress test.

**Pull request closed without merging**